examples/extensions-zalando/src/main/java/org/zalando/intellij/swagger/examples/extensions/zalando/validator/zally/YamlFileValidator.java

```diff
@@ -0,0 +1,89 @@
+package org.zalando.intellij.swagger.examples.extensions.zalando.validator.zally;
+
+import com.google.common.collect.Lists;
+import com.intellij.codeInspection.*;
+import com.intellij.openapi.components.ServiceManager;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.yaml.psi.YAMLKeyValue;
+import org.zalando.intellij.swagger.examples.extensions.zalando.validator.ZallySettings;
+import org.zalando.intellij.swagger.examples.extensions.zalando.validator.zally.model.LintingResponse;
+import org.zalando.intellij.swagger.examples.extensions.zalando.validator.zally.model.Violation;
+import org.zalando.intellij.swagger.file.FileDetector;
+import org.zalando.intellij.swagger.traversal.path.PathFinder;
+
+import java.util.List;
+import java.util.Optional;
+
+public class YamlFileValidator extends LocalInspectionTool {
+
+    private final static Logger LOG = Logger.getInstance(YamlFileValidator.class);
+
+    private final PathFinder pathFinder = new PathFinder();
+    private final FileDetector fileDetector = new FileDetector();
+
+    @Override
+    public ProblemDescriptor[] checkFile(@NotNull PsiFile file, @NotNull InspectionManager manager, boolean isOnTheFly) {
+        if (shouldValidate(file)) {
+            final LintingResponse lintingResponse = validate(file);
+            return createViolations(manager, isOnTheFly, lintingResponse, file);
+        }
+
+        return ProblemDescriptor.EMPTY_ARRAY;
+    }
+
+    private boolean shouldValidate(final PsiFile file) {
+        return hasZallyUrl() && isSwaggerFile(file);
+    }
+
+    private boolean isSwaggerFile(PsiFile file) {
+        return fileDetector.isMainSwaggerYamlFile(file);
+    }
+
+    private boolean hasZallyUrl() {
+        final ZallySettings zallySettings = ServiceManager.getService(ZallySettings.class);
+
+        return zallySettings.zallyUrl != null && !zallySettings.zallyUrl.isEmpty();
+    }
+
+    private LintingResponse validate(final PsiFile file) {
+        try {
+            final ZallyService zallyService = ServiceManager.getService(ZallyService.class);
```

Comment:
You create a Zally service every time the `validate` function is called. Let's do it once, during the initialization.

Reply:
I agree, this makes the class cleaner (although an instance is created once when using the `ServiceManager`), fixed in https://github.com/zalando/intellij-swagger/pull/188/commits/255c734bd9ddba0efc19c23568810e79118d449e.